src/docs/devdoc-schema.dox

```diff
@@ -0,0 +1,116 @@
+/*! @page devdoc-schema Schema operations
+
+A schema defines the format of the application data in WiredTiger. WiredTiger
+supports various types of schemas (See @ref schema for more
+information), operated upon through a WT_SESSION reference. This section details
+the internals of these various schema operations.
+
+Schema operations cause an update to the metadata and are performed under a
+schema lock to avoid concurrent operations on the same object. Following
+sequence of steps define a generic schema operation:
+
+@plantuml_start{schema_generic.png}
+@startuml{schema_generic.png}
+:WT_SESSION->generic_schema_op;
+partition with-schema-lock {
+    :perform operation on underlying data-object;
+    :update metadata-file;
+    :checkpoint and sync metadata;
+}
+stop
+@enduml
+@plantuml_end
+
+@section schema_create Schema Create
+
+The create schema operation is responsible for creating the underlying data
+object on the filesystem with the right parameters and then creating an entry
+for this new object into the metadata. The sequence of operations for various
+schema types are as follows:
+
+@plantuml_start{schema_create.png}
+@startuml{schema_create.png}
+:WT_SESSION->create
+(__session_create());
+
+partition session-API-call {
+    :API session init with NULL dhandle;
+    :exit if PANIC flag set;
+    :exit if invalid configuration;
+}
+
+:validate "type" config parameter;
+
+partition with-schema-lock {
+    partition with-table-lock {
+        :turn on meta tracking;
+        :check uri}
+
+        split
+            :uri matches "colgroup:";
+            :__create_colgroup();
+        split again
+            :uri matches "file:";
```

Comment:
I'd restructure this a little bit. It's generally not expected that users operate directly on _file:_ URIs - those are pseudo internal, and used as building blocks for other data types. I'd create a picture with the four other URI types you identified, and have them lead into the file block. Some of them should have a many-to-one line to the file block (as in a table can be made up of many files).